htdocs/core/tpl/extrafields_list_search_sql.tpl.php

```diff
@@ -35,7 +35,8 @@
 		{
 			$mode_search = 0;
 			if (in_array($typ, array('int', 'double', 'real', 'price'))) $mode_search = 1; // Search on a numeric
-			if (in_array($typ, array('sellist', 'link')) && $crit != '0' && $crit != '-1') $mode_search = 2; // Search on a foreign key int
+			if (in_array($typ, array('sellist', 'link')) && $crit != '0' && $crit != '-1' && is_numeric($crit)) $mode_search = 2; // Search on a foreign key int
+			if (in_array($typ, array('sellist')) && is_string($crit)) $mode_search = 0;// Search on a foreign key string
```

Comment:
same here, should be !is_numeric($crit)  ?

Reply:
We want $mode_search=2 when $crit is a number, rowid or forgin key. 
If I set a extrafields select from table with table:code:rowid, the value store will be the int rowid => search_mode = 2
If I set a extrafields select from table with table:label:code, the value store will be the alphanumeric code and not the rowid.   => search_mode = 0